server/controllers/userController.js

```diff
@@ -1,112 +1,157 @@
+    res.status(500).json({ message: "Server error" });
   }
-
-  const updatedUser = await User.findOneAndUpdate(
-    { email },
-    { $push: { savedPlans: newPlan } },
-    { new: true }
-  );
-
-  res
-    .status(201)
-    .json({ message: "Plan added to saved plans", user: updatedUser });
 };
 
 const updatePlanUser = async (req, res) => {
-  const { email, newPlan } = req.body;
+  try {
+    const { email } = req.params;
+    const { newPlan } = req.body;
 
-  const user = await User.findOne({ email });
+    const user = await User.findOne({ email });
 
-  if (!user) {
-    return res.status(404).json({ message: "User not found" });
-  }
+    if (!user) {
+      return res.status(404).json({ message: "User not found" });
+    }
 
-  console.log(email);
-  const planIndex = user.savedPlans.findIndex(
-    (plan) => plan.name === newPlan.name
-  );
+    console.log(email);
 
-  if (planIndex === -1) {
-    return res.status(404).json({ message: "Plan not found" });
-  }
+    const planIndex = user.savedPlans.findIndex(
+      (plan) => plan.name === newPlan.name,
+    );
+
+    if (planIndex === -1) {
+      return res.status(404).json({ message: "Plan not found" });
+    }
 
-  user.savedPlans[planIndex] = newPlan;
+    user.savedPlans[planIndex] = newPlan;
 
-  const updatedUser = await User.findOneAndUpdate(
-    { email },
-    { savedPlans: user.savedPlans },
-    { new: true }
-  );
+    const updatedUser = await User.findOneAndUpdate(
+      { email },
+      { savedPlans: user.savedPlans },
+      { new: true },
+    );
 
-  res
-    .status(200)
-    .json({ message: "Plan updated in saved plans", user: updatedUser });
+    res
+      .status(200)
+      .json({ message: "Plan updated in saved plans", user: updatedUser });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Server error" });
+  }
 };
 
 const getUserPlans = async (req, res) => {
-  const { email } = req.body;
+  try {
+    const { email } = req.params;
 
-  const user = await User.findOne({ email });
+    if (!email) {
+      return res.status(400).json({ message: "Email is required " });
+    }
 
-  if (!user) {
-    return res.status(404).json({ message: "User not found" });
+    const user = await User.findOne({ email });
+
+    if (!user) {
+      return res.status(404).json({ message: "User not found" });
+    }
+
+    return res.status(200).json(user.savedPlans);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Server error" });
   }
-  return res.status(200).json(user.savedPlans);
 };
 
 const deleteUserPlan = async (req, res) => {
-  const { email, planName } = req.body;
-
-  const user = await User.findOneAndUpdate(
-    { email },
-    { $pull: { savedPlans: { name: planName } } },
-    { new: true }
-  );
-
-  if (!user) {
-    return res.status(404).json({ message: "User not found" });
+  try {
+    const { email } = req.params;
+    const { planName } = req.body;
```

Comment:
I think you get the ``planName`` from ``req.params['planName']`` and not ``req.body``